libknet/links_acl_ip.c

```diff
@@ -279,10 +279,15 @@ int ipcheck_addip(void *fd_tracker_match_entry_head, int index,
 			/*
 			 * find the end of the list or stop at "index"
 			 */
-			while ((match_entry->next) || (i < index)) {
+
+			while (match_entry->next) {
```

Comment:
Wouldn't `while ((match_entry->next) && (i < index)) {` (or maybe `(i <= index)`, I'm not sure what is the intended result) work same way?

Reply:
I have tried different combinations including the ones you suggested and somehow it was failing to pass the test suite because it was adding the new ACL in the wrong place.

This code is not on the critical performance path either, so i just decided to keep it simpler and loop only against one variable.

Reply:
Ok, so it's because index is -1. I don't agree that code is really simpler (it would be simpler with list abstraction, as for example libqb list, queue.h, ...), but this seems to work too.